File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        int product=1, sum=0, j=1;
        String num = Integer.toString(n);
        for(int i=0; i<num.length();i++){
           j = Character.getNumericValue(num.charAt(i));
           product*=j;
           sum+=j;
        }
        return product-sum;   
    }
}

File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function(n) {
    let product=1,sum=0;
    for(let digit of digits(n)){
        product*=digit;
        sum+=digit;
    }
    return product-sum;
    
};

function* digits(n){
    while(n>0){
        yield n%10;
        n=Math.floor(n/=10);
    }
}